csvcubed/csvcubed/readers/v1_0/columnschema.py

```diff
@@ -0,0 +1,459 @@
+                and self.definition_uri.lower().startswith("http://reference.data.gov.uk/id/")
+            ):
+                # This is a special case where we build up a code-list of the date/time values.
+                code_list_obj = self._get_date_time_code_list_for_dimension(
+                    self.label, new_dimension
+                )
+                # else, the user wants a standard codelist to be automatically generated
+                return code_list_obj or new_dimension.code_list
+        else:
+            raise ValueError(f"Unmatched code_list value {self.code_list}")
+
+        return code_list_obj or new_dimension.code_list
+
+    def _get_date_time_code_list_for_dimension(
+        self, column_title: str, new_dimension: NewQbDimension
+    ) -> CompositeQbCodeList:
+        csvw_safe_column_title = csvw_column_name_safe(column_title)
+        assert isinstance(new_dimension.code_list, NewQbCodeList)
+        return CompositeQbCodeList(
+            CatalogMetadata(new_dimension.label),
+            [
+                DuplicatedQbConcept(
+                    existing_concept_uri=uritemplate.expand(
+                        c.value,
+                        {csvw_safe_column_title: c.label},
+                    ),
+                    label=c.label,
+                    code=c.code,
+                )
+                for c in new_dimension.code_list.concepts
+            ],
+        )
+
+
+@dataclass
+class ExistingDimension(SchemaBaseClass):
+    from_existing: str = ""
+    cell_uri_template: Optional[str] = None
+
+    def map_to_existing_qb_dimension(
+        self, label: str, data: PandasDataTypes, json_parent_dir: Path
+    ) -> ExistingQbDimension:
+        return ExistingQbDimension(self.from_existing)
+
+
+@dataclass
+class AttributeValue(SchemaBaseClass):
+    label: str
+    description: Optional[str] = None
+    from_existing: Optional[str] = None
+    definition_uri: Optional[str] = None
+
+
+@dataclass
+class NewAttributeProperty(SchemaBaseClass):
+    path: Optional[str] = None
+    label: Optional[str] = None
+    comment: Optional[str] = None
+    isDefinedBy: Optional[str] = None
+    subPropertyOf: Optional[str] = None
+    newAttributeValues: Union[None, bool, List[AttributeValue]] = None
+    literalValuesDataType: Optional[str] = None
+
+
+@dataclass
+class ExistingAttribute(SchemaBaseClass):
+    from_existing: str = ""
+    definition_uri: Optional[str] = None
+    data_type: Optional[str] = None
+    required: bool = False
+    values: Union[bool, List[AttributeValue]] = True
+
+    def map_to_qb_attribute(
+        self,
+        label: str,
+        data: PandasDataTypes
+    ) -> ExistingQbAttribute:
+        return ExistingQbAttribute(
+            self.from_existing,
+            new_attribute_values=_get_new_attribute_values(
+                data, self.values
+            ),
+            is_required=self.required,
+        )
+
+    def map_to_existing_qb_attribute(
```

Comment:
I don't see why you have two functions here, but this one seems like the better function to call. You should probably get rid of `map_to_qb_attribute`.

Reply:
removed, it was left-over from the combined New/Existing code.